src/Microsoft.Dnx.TestHost/Program.cs

```diff
@@ -32,16 +30,47 @@ public int Main(string[] args)
 
             var env = (IApplicationEnvironment)_services.GetService(typeof(IApplicationEnvironment));
 
-            var portOption = application.Option("--port", "Port number to listen for a connection.", CommandOptionType.SingleValue);
-            var projectOption = application.Option("--project", "Path to a project file.", CommandOptionType.SingleValue);
+            var portOption = application.Option(
+                "--port", "Port number to listen for a connection.", CommandOptionType.SingleValue);
+
+            var projectOption = application.Option(
+                "--project", "Path to a project file.", CommandOptionType.SingleValue);
 
             var debugOption = application.Option("--debug", "Launch the debugger", CommandOptionType.NoValue);
 
             var waitOption = application.Option("--wait", "Wait for attach", CommandOptionType.NoValue);
 
+            var parentProcessIdOption = application.Option(
+                "--parentProcessId",
+                "Process id of the parent which launched this process",
+                CommandOptionType.SingleValue);
+
             // If no command was specified at the commandline, then wait for a command via message.
             application.OnExecute(async () =>
             {
+                // Register for parent process's exit event
+                var parentProcessId = parentProcessIdOption.Value();
+                if (!string.IsNullOrEmpty(parentProcessId))
+                {
+                    var parentProcess = Process.GetProcesses()
+                        .FirstOrDefault(p => p.Id == Convert.ToInt32(parentProcessId));
```

Comment:
Don't parse the integer inside the query. Should we do something to validate that it can be converted?